### ISSUE TITLE
Namespace conflict issue after 2.2.0

### ISSUE THREAD

**Issue by shrkw** [socketry/multipart-post#98]
This is not global issue, maybe a kind of case report.

2.2.0 contains big refactoring (https://github.com/socketry/multipart-post/pull/65) and it makes sense. However compatibility code made an issue on my code.

I have `Parts` module,

```
module Parts
  class Foo; end
end
```

And with 2.1.1, it works fine. 

```
$ bundle exec irb
2.7.2 :001 > require_relative 'parts/foo'
 => true 
2.7.2 :002 > require 'net/http/post/multipart'
 => true 
2.7.2 :003 > Parts::Foo.class
 => Class 
2.7.2 :004 > Parts.constants
 => [:Foo, :Part, :EpiloguePart, :FilePart, :ParamPart] 
```

However, with 2.2.0, it have loading order issue.

When load modules on following order, it works.

1. Multipart gem
2. my `Parts` module

```
2.7.2 :001 > require 'net/http/post/multipart'
 => true 
2.7.2 :003 > Multipart::Post::Parts.class
 => Module 
2.7.2 :006 > require_relative 'parts/foo'
 => true 
2.7.2 :007 > Parts.class
 => Module 
2.7.2 :011 >  Parts.constants
 => [:Foo] 
```

But when following order, it does not work.

1. my `Parts` module
2. Multipart gem

```
2.7.2 :001 > require_relative 'parts/foo'
 => true 
2.7.2 :002 > require 'net/http/post/multipart'
/usr/local/rvm/gems/ruby-2.7.2/gems/multipart-post-2.2.3/lib/multipart/post/parts.rb:151: warning: already initialized constant Parts
/workspaces/sandbox-ruby/simple/parts/foo.rb:3: warning: previous definition of Parts was here
 => true                                                            
2.7.2 :003 > Parts::Foo
Traceback (most recent call last):
        21: from /usr/local/rvm/gems/ruby-2.7.2/bin/ruby_executable_hooks:22:in `<main>'
        20: from /usr/local/rvm/gems/ruby-2.7.2/bin/ruby_executable_hooks:22:in `eval'
        19: from /usr/local/rvm/gems/ruby-2.7.2/bin/irb:23:in `<main>'
        18: from /usr/local/rvm/gems/ruby-2.7.2/bin/irb:23:in `load'
        17: from /usr/local/rvm/gems/ruby-2.7.2/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
(irb):3:in `<main>': uninitialized constant Multipart::Post::Parts::Foo (NameError)
```

I guess it led by here https://github.com/socketry/multipart-post/blob/31500a7008d0e5827c8522d6d1bde3452237b578/lib/multipart/post/parts.rb#L151 , and it is for compatibility.

If that part of code change, it would be breaking change. so, I'm not sure how it should be fixed.

Do you have any ideas? Like applying a monkey patch?

**Comment by ioquatix** [comment]
Can you try main branch and see if it fixes your issue?

**Comment by shrkw** [comment]
The issue was fixed on main branch!

```Gemfile
gem "multipart-post", git: 'git@github.com:socketry/multipart-post.git', ref: 'main'
```

```
2.7.2 :001 > require_relative 'parts/foo'
2.7.2 :002 > require 'net/http/post/multipart'
 => true 
2.7.2 :003 > Parts.constants
 => [:Foo] 
 => Class 
2.7.2 :008 > Net::HTTP::Post::Multipart.class
 => Class 
```

Thank you so much!

**Comment by ioquatix** [comment]
I guess we can fix this in a minor version bump. The warnings have been in place for a while now.

**Comment by robotfelix** [comment]
I know it's fixed already on the `main` branch, but thought it's worth reporting that I've run into this bug too, and will look forward to the next release!

**Comment by ioquatix** [comment]
I'll cut a new release soon!, sorry.

**Comment by ioquatix** [comment]
Released in v2.3.0.
